src/app/contexts/RequestContext/index.jsx

```diff
@@ -17,7 +17,7 @@ export const RequestContextProvider = ({
   previousPath,
 }) => {
   const { isUK, origin } = getOriginContext(bbcOrigin);
-  const env = getEnv(origin);
+  const env = isLocal ? 'local' : getEnv(origin);
```

Comment:
I don't quite follow why we need to do this. `getEnv` should return `local` if the origin includes the string `local`:
https://github.com/bbc/simorgh/pull/2815/files#diff-75a773bc6e656cc4f1f782712cd28c7eR2-R4

But at this level we're now looking at the output of `isLocal` (based on `process.env`) to override that value, so the logic seems split between two different places. Wouldn't it make more sense for it all to be in `getEnv`?

Reply:
>getEnv should return local if the origin includes the string local

The server render doesnt have an origin locally

Reply:
Ah thanks @dr3, just to make sure we're clear, you're saying that in local development `bbcOrigin` is undefined here?

https://github.com/bbc/simorgh/blob/0aaaf30f5269c3b920375ec33b6ea26f62a4eae9/src/server/index.jsx#L173

In that case what is the purpose of this?

https://github.com/bbc/simorgh/blob/26ad706388e2605639944c5ac8187408e7679cd3/src/app/contexts/RequestContext/getEnv/index.js#L2-L4

Reply:
Yep, locally `bbcOrigin` is undefined, so https://github.com/bbc/simorgh/blob/latest/src/app/contexts/RequestContext/getOriginContext/index.js#L10 sets origin to localhost 

And then getEnv reads origin as localhost here https://github.com/bbc/simorgh/blob/26ad706388e2605639944c5ac8187408e7679cd3/src/app/contexts/RequestContext/getEnv/index.js#L2-L4

Reply:
https://github.com/bbc/simorgh/pull/2815#issuecomment-516851453 :)